test/spec/services/groupServicesSpec.js

```diff
@@ -16,27 +17,121 @@ describe('groupServices', function() {
     });
   });
 
+  it('calls get groups', function() {
+    $httpBackend.expectGET(groupApiUrlPrefix).respond({});
+    groups.Group.getGroups();
+    $httpBackend.flush();
+  });
+
+  it('calls get group', function() {
+    var gid = 1;
+    $httpBackend.expectGET(groupApiUrlPrefix + '/' + gid).respond({});
+    groups.Group.get(gid);
+    $httpBackend.flush();
+  });
+
   it('calls party endpoint', function() {
-    $httpBackend.expectGET('/api/v2/groups/party').respond({});
-    groups.party();
+    $httpBackend.expectGET(groupApiUrlPrefix + '/party').respond({});
+    groups.Group.syncParty();
+    $httpBackend.flush();
+  });
+
+  it('calls create endpoint', function() {
+    $httpBackend.expectPOST(groupApiUrlPrefix).respond({});
+    groups.Group.create({});
+    $httpBackend.flush();
+  });
+
+  it('calls update group', function() {
+    var gid = 1;
+    var groupDetails = { _id: gid };
+    $httpBackend.expectPUT(groupApiUrlPrefix + '/' + gid).respond({});
+    groups.Group.update(groupDetails);
+    $httpBackend.flush();
+  });
+
+  it('calls join group', function() {
+    var gid = 1;
+    $httpBackend.expectPOST(groupApiUrlPrefix + '/' + gid + '/join').respond({});
+    groups.Group.join(gid);
+    $httpBackend.flush();
+  });
+
+  it('calls reject invite group', function() {
+    var gid = 1;
+    $httpBackend.expectPOST(groupApiUrlPrefix + '/' + gid + '/reject-invite').respond({});
+    groups.Group.rejectInvite(gid);
     $httpBackend.flush();
   });
 
-  it('calls tavern endpoint', function() {
-    $httpBackend.expectGET('/api/v2/groups/habitrpg').respond({});
+  it('calls invite group', function() {
+    var gid = 1;
+    $httpBackend.expectPOST(groupApiUrlPrefix + '/' + gid + '/invite').respond({});
+    groups.Group.invite(gid, [], []);
+    $httpBackend.flush();
+  });
+
+  it('calls tavern endpoint when tavern is not cached', function() {
+    $httpBackend.expectGET(groupApiUrlPrefix + '/habitrpg').respond({});
     groups.tavern();
     $httpBackend.flush();
   });
 
+  it('returns tavern if cached', function (done) {
+    var uid = 'abc';
+    var tavern = {
+      _id: uid,
+    };
+    groups.data.tavern = tavern;
+    groups.tavern()
+      .then(function (tavern) {
+        expect(tavern).to.eql(tavern);
```

Comment:
This is not testing anything since the `tavern` variable in this function will reference the argument in the function, and not the variable you instantiated earlier. Should be something like this:

groups.tavern().then(function (result) {
  expect(result).to.eql(tavern);
});